### PR TITLE
Support Thrift 0.9.3 and 0.10.0

### LINKED ISSUE #387: Update Apache Thrift to 0.10.0

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

Thrift 0.10.0 has been released with some breaking changes. We should
support both Thrift 0.9 and 0.10 until majority of our users finishes
migration.

Modifications:

- Make 'armeria-thrift' depend on libthrift 0.10.0
  - Use `TBase<?, ?>` because of the type parameter signature changes of
    TBase in 0.10.0
  - Decode TApplicationException using internal.thrift.TApplicationExceptions
    to work around the API breakage in TApplicationException.read()
    between 0.9.3 and 0.10.0
  - Add a few utility methods to ThriftFieldAccess
- Add a new project 'armeria-thrift0.9' that depends on libthrift 0.9.3
  - It does not contain any Java source code. It uses the source code
    from the 'armeria-thrift' project, only resulting different byte
    code due to the ABI compatibility issues between 0.9 and 0.10.
  - It contains Thrift source files because we replaced 'byte' with 'i8'
    in 'armeria-thrift' and Thrift 0.9 does not recognize 'i8'.
- Remove the dependency to 'armeria-thrift' from 'armeria-kafka'

Result:

- A user can choose between Thrift 0.9 and 0.10 by depending on
  'armeria-thrift0.9' or 'armeria-thrift'.
- Fixes #387

### CODE REVIEW

File: thrift0.9/src/test/thrift/RpcDebugTest.thrift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace java com.linecorp.armeria.common.thrift.text
```
Comment (trustin) [inline thrift0.9/src/test/thrift/RpcDebugTest.thrift]: This file has been copied from `thrift/src/test/thrift`.

File: thrift0.9/src/test/thrift/TTextProtocolTest.thrift [hunk 2]
```diff
@@ -0,0 +1,102 @@
+// =================================================================================================
```
Comment (trustin) [inline thrift0.9/src/test/thrift/TTextProtocolTest.thrift]: This file has been copied from `thrift/src/test/thrift`.

File: thrift0.9/src/test/thrift/main.thrift [hunk 3]
```diff
@@ -0,0 +1,112 @@
+namespace java com.linecorp.armeria.service.test.thrift.main
```
Comment (trustin) [inline thrift0.9/src/test/thrift/main.thrift]: This file has been copied from `thrift/src/test/thrift`.

**Comment by trustin** [comment]
One to-do item left: remove the thrift dependency from `armeria-kafka` so that a `armeria-thrift0.9` user has no issues.

**Comment by trustin** [comment]
Thanks for reviewing, @imasahiro. Please feel free to merge this if you think it LGTM. :-)

**Comment by imasahiro** [comment]
Nice work @trustin !

**Comment by trustin** [comment]
Thank you, @imasahiro :-)

**Pull request merged**